src/tests.rs

```diff
@@ -0,0 +1,185 @@
+
+            let subxt_block_number: subxt::BlockNumber =
+                signed_commitment.commitment.block_number.into();
+            let block_hash = client
+                .rpc()
+                .block_hash(Some(subxt_block_number))
+                .await
+                .unwrap();
+
+            let current_authorities = api.storage().beefy().authorities(block_hash).await.unwrap();
+
+            // Current LeafIndex
+            let block_number = signed_commitment.commitment.block_number;
+            let leaf_index = (block_number - 1) as u64;
+            let leaf_proof: pallet_mmr_rpc::LeafProof<H256> = client
+                .rpc()
+                .client
+                .request("mmr_generateProof", rpc_params!(leaf_index, block_hash))
+                .await
+                .unwrap();
+
+            let opaque_leaf: Vec<u8> = codec::Decode::decode(&mut &*leaf_proof.leaf.0).unwrap();
+            let latest_leaf: MmrLeaf<u32, H256, H256> =
+                codec::Decode::decode(&mut &*opaque_leaf).unwrap();
+            let mmr_proof: pallet_mmr_primitives::Proof<H256> =
+                codec::Decode::decode(&mut &*leaf_proof.proof.0).unwrap();
+
+            let authority_address_hashes = current_authorities
+                .into_iter()
+                .map(|x| {
+                    let id: beefy_primitives::crypto::AuthorityId =
+                        codec::Decode::decode(&mut &*x.encode()).unwrap();
+                    keccak_256(&beefy_mmr::BeefyEcdsaToEthereum::convert(id))
+                })
+                .collect::<Vec<_>>();
+
+            let signatures = signed_commitment
+                .signatures
+                .into_iter()
+                .map(|x| {
+                    if let Some(sig) = x {
+                        let mut temp = [0u8; 65];
+                        if sig.len() == 65 {
+                            temp.copy_from_slice(&*sig.encode());
+                            Some(temp)
+                        } else {
+                            None
+                        }
+                    } else {
+                        None
+                    }
+                })
+                .collect::<Vec<_>>();
+
+            let signature_indices = signatures
+                .iter()
+                .enumerate()
+                .filter_map(|x| if x.1.is_some() { Some(x.0) } else { None })
+                .collect::<Vec<_>>();
+
+            let tree =
+                rs_merkle::MerkleTree::<KeccakHasher>::from_leaves(&authority_address_hashes);
+
+            let authority_proof = tree.proof(&signature_indices);
+
+            let mmr_update = MmrUpdateProof {
+                signed_commitment: SignedCommitment {
+                    commitment: signed_commitment.commitment,
+                    signatures,
+                },
+                latest_mmr_leaf_with_index: MmrLeafWithIndex {
+                    leaf: latest_leaf,
+                    index: leaf_index,
```

Comment:
Let's remove this, we can infer leaf index from the block number in the MMR leaf